bob-apps/bin/bobd.rs

```diff
@@ -139,6 +169,74 @@ async fn main() {
     }
 }
 
+fn configure_testmode(sub_matches: &ArgMatches) -> AnyResult<(ClusterConfig, NodeConfig)> {
+    let mut addresses = Vec::with_capacity(1);
+    let port = match sub_matches.value_of("grpc-port") {
+        Some(v) => v.parse().context("could not parse --grpc-port")?,
+        None => 20000
+    };
+    let mut this_node = None;
+    if let Some(node_list) = sub_matches.value_of("nodes") {
+        let available_ips: HashSet<String> = NetworkInterface::show()?.into_iter().filter_map(|itf|
+            match itf.addr? {
+                Addr::V4(addr) => {
+                    Some(addr.ip.to_string())
+                },
+                _ => None
+        }).collect();
+
+        for (index, addr) in node_list.split(",").enumerate() {
+            let split = &addr.split_once(":").context("expected nodes address to have format ip:port")?;
+            let in_ip = split.0;
+            let in_port = split.1.parse::<u16>().context("failed to parse port")?;
```

Comment:
Looks like it would be simpler to parse value into [SocketAddrV4](https://doc.rust-lang.org/nightly/core/net/struct.SocketAddrV4.html#). Also for simplicity it would be good to collect `Ipv4Addr` above from `NetworkInterface::show()` (not perform `to_string()` conversion)